src/user/room.ts

```diff
@@ -594,6 +645,38 @@ export class Room extends Accessory implements Sayable {
   public async announce()             : Promise<string>
   public async announce(text: string) : Promise<void>
 
+  /**
+   * SET/GET announce from the room
+   * > Tips: It only works when bot is the owner of the room.
+   *
+   * @param {string} [text] If set this para, it will change room announce.
+   * @returns {(Promise<void | string>)}
+   *
+   * @example <caption>When you say anything in a room, it will get room announce. </caption>
```

Comment:
Do not use this complicated example, because the room announces operation is no related with the `message` event.

Just find a room, and get/set the announcement.